### DISCUSSION TITLE
Log to a file

### DISCUSSION THREAD

**Discussion by bmccorm2** [nginx-proxy/nginx-proxy#2190] [Q&A] [answered]
Sorry for noob question, but it seems the logs go to stdout, can i redirect them to a log file instead?

**Comment by rhansen** [comment, marked as answer]
Which logs are you referring to?

  * nginx access logs already go to a file (`/var/log/nginx/access.log`)
  * nginx error logs and docker-gen logs go to stderr (not currently configurable), which should be captured in `docker logs`.
  * nginx-proxy startup messages go to a mix of stderr and stdout depending on severity (not currently configurable), which should be captured in `docker logs`.

**Reply by bmccorm2** [reply]
Thanks.  I think this corresponds to your point # 2 - currently my logs point to stdout and stderr and you said this was not configurable.  So there is no way to create a volume and get these logs on the host machine?  Thanks!

```
/var/log/nginx # ls -l
total 0
lrwxrwxrwx    1 root     root            11 Feb 11 03:03 access.log -> /dev/stdout
lrwxrwxrwx    1 root     root            11 Feb 11 03:03 error.log -> /dev/stderr
```

**Reply by SchoNie** [reply]
If you want to capture access.log and error.log from nginx, that corresponds to point # 1 from rhansen.
You can bind mount a local folder to `/var/log/nginx/` and the logs will be written to a persistent location.

You can add a volume to your docker compose. (are you using compose? if you post your complete file I can help to adjust)
```
    volumes:
      - "./accesslogs/:/var/log/nginx/"
```

Then the access.log + error.log and written to "accesslogs" subfolder relative to the docker-compose.yml
Make sure you do some kind of [logrotation ](https://linux.die.net/man/8/logrotate) to prevent from filling up your disk.

**Reply by bmccorm2** [reply]
Cool thanks.  I looked at those files in the container and they were blank (because they were redirected to stdout) - but when i mounted a volume through docker-compose it worked great.  Thanks again!
